base/abstractarray.jl

```diff
@@ -251,8 +251,8 @@ julia> strides(A)
 """
 function strides end
 
-# the definition of strides for Array is the cumsum of the product of sizes
-function _cumsumprodsizes(a::AbstractArray, i)
+# the definition of strides for Array is cumprod([1;collect(size(A))[1:end-1]])
```

Comment:
well, technically `cumprod` would return an array, whereas strides returns a tuple

Reply:
Hmm, that's not quite right since:
```julia
julia> A = Array{Float64,0}()
0-dimensional Array{Float64,0}:
0.0

julia> strides(A)
()
```
How about saying:

> the defition of strides for `Array{T,N}` is `tuple()` if `N = 0`, otherwise it is a tuple containing 1 and a cumulative product of the first N-1 sizes